twisted/web/test/test_web.py

```diff
@@ -136,6 +136,32 @@ def test_makeSession(self):
         self.assertIsInstance(session.uid, bytes)
 
 
+    def test_sessionUIDGeneration(self):
+        """
+        L{site.getSession} generates L{Session} objects with distinct UIDs from
+        a secure source of entropy.
+        """
+        site = server.Site(resource.Resource())
+        # Ensure that we _would_ use the unpredictable random source if the
+        # test didn't stub it.
+        self.assertIdentical(site._entropy, os.urandom)
+
+        def predictableEntropy(n):
+            predictableEntropy.x += 1
+            return (unichr(predictableEntropy.x) * n).encode("charmap")
+        predictableEntropy.x = 0
+        self.patch(site, "_entropy", predictableEntropy)
+        a = self.getAutoExpiringSession(site)
+        b = self.getAutoExpiringSession(site)
+        self.assertEqual(a.uid, b"01" * 0x20)
```

Comment:
Taken under advisement for future reference; but I'd rather not wait for another full CI circuit to land :-)